Handle special keys

Metalgrid opened the pull request:
Correctly send special keys (*arrow keys, insert, delete, end, home, page up, page down, etc*), which send an escape sequence and a keycode from a single key press (e.g. *up-arrow* sends `^[[A`). This fixes *vim* running in an interactive terminal session, but requires to put `sys.stdin` in non-blocking mode so that regular keys are sent without filling a buffer.

bitprophet commented:
Thanks for this. I like the idea of making nontrivial demo sessions work better, though I wonder whether this could increase chances of the base case breaking for some users?

Guessing not (I can't think of many situations offhand where a nonblocking stdin would make somebody's terminal wig out) but something I'm always wary of.

Maybe some additional comments noting what the `fcntl` calls are doing & how best to remove them if they're problematic, would be the best middle road.

Metalgrid commented:
I removed the non-blocking code and switched to `os.read()` on `sys.stdin`'s file descriptor instead. So far it works just as good without using non-blocking mode.

flobz commented:
I can confirm that it fixed #1494.
Can we merge please ?
